scipy/spatial/distance.py

```diff
@@ -1395,6 +1451,10 @@ def dfun(u, v):
 
     elif isinstance(metric, string_types):
         mstr = metric.lower()
+        if w is not None and not mstr.startswith("test_"):
+            if mstr in ['seuclidean', 'se', 's', 'mahalanobis']:
+                raise ValueError("metric %s incompatible with weights" % mstr)
+            mstr = "test_%s" % mstr
```

Comment:
IMHO, not very clear. Also, it seems to me that it may lead to different behaviour between the "test" and the regular implementation